src/monitor/node_active_protocol.c

```diff
@@ -1259,6 +1260,120 @@ perform_failover(PG_FUNCTION_ARGS)
 }
 
 
+/*
+ * promote promotes a given target node in a group.
+ */
+Datum
+perform_promotion(PG_FUNCTION_ARGS)
+{
+	text *formationIdText = PG_GETARG_TEXT_P(0);
+	char *formationId = text_to_cstring(formationIdText);
+
+	text *nodeNameText = PG_GETARG_TEXT_P(1);
+	char *nodeName = text_to_cstring(nodeNameText);
+
+	List *groupNodesList = NULL;
+	int totalNodesCount = 0;
+
+	AutoFailoverNode *currentNode = NULL;
+
+	checkPgAutoFailoverVersion();
+
+	currentNode = GetAutoFailoverNodeByName(formationId, nodeName);
```

Comment:
If the node is  in maintenance (or unhealthy?), it is not/may not  get promoted.

In MS case, we could promote another  node, in single standby we error out as there is no primary but might not  be very clear to the users `Monitor ERROR:  couldn't find the primary node in formation "default", group 0`


So, maybe we give a proper error here if the node unhealthy AND in maintenance mode?

Reply:
I think it's fine to re-use error message from the perform failover code path. The intention is to show in the docs that the `pg_autoctl perform promotion` is a special case of the `pg_autoctl perform switchover` command.

Reply:
I agree that error messages should come from failover code. But, the following behavior is confusing:
- node 1: primary, node2: secondary node3: maintenance
- perform promotion on node 3
- node2 becomes the primary as failover code would ignore node3

This seems like a new case that failover code is not aware of?

Reply:
Oh I see what you mean, we could refuse the target node for `perform_promotion` when it's not currently in the SECONDARY state. That makes sense, going to add this test.

Reply:
Added now!